tests/DateTimeTest.php

```diff
@@ -2,12 +2,84 @@
 
 namespace Psalm\Tests;
 
+use Exception;
+use Psalm\Context;
 use Psalm\Tests\Traits\ValidCodeAnalysisTestTrait;
 
+use function version_compare;
+
+use const PHP_VERSION;
+
 class DateTimeTest extends TestCase
 {
     use ValidCodeAnalysisTestTrait;
 
+    public function testModifyWithInvalidConstant(): void
+    {
+        $context = new Context();
+
+        if (version_compare(PHP_VERSION, '8.3', '>')) {
```

Comment:
why 8.3 is excluded? Or did you mean `>= 8.3`? Again, please use comparison with `PHP_VERSION_ID` instead.

Reply:
With PHP 8.3 DateTime modify() Warning was converted to DateMalformedStringException, e.g.

php8.2 -r "(new DateTime())->modify('foo');"
PHP Warning:  DateTime::modify(): Failed to parse time string (foo) at position 0 (f): The timezone could not be found in the database in Command line code on line 1

php8.3 -r "(new DateTime())->modify('foo');"
PHP Fatal error:  Uncaught DateMalformedStringException: DateTime::modify(): Failed to parse time string (foo) at position 0 (f): The timezone could not be found in the database in Command line code:1
Stack trace:
#0 Command line code(1): DateTime->modify()
#1 {main}
  thrown in Command line code on line 1

Reply:
Using PHP_VERSION_ID is ok.